// login-query-express-api/extensions.js
export const homeHandler = (req, res) => {
  res
    .status(500)
    .send(
      `<h1>Welcome to our fully functional <span style="color: green;">API</span>!</h1>`
    );
};
export const nameExtension = (req, res) => {
  const name = req.params.name;
  res
    .status(500)
    .send(
      `<h1>Welcome <span style="color: green;">${name}</span>! Happy to have you again!😄</h1>`
    );
};

export const searchExtension = (req, res) => {
  const keyword = req.query.keyword || "something";

  res.send(`
    <!DOCTYPE html>
    <html lang="en">
    <head>
      <meta charset="UTF-8">
      <title>Searching...</title>
      <style>
        .highlight{
            color: green;
        }
        .loading-dots {
          display: inline-block;
          min-width: 3ch;
        }
        
        .dot {
          animation: blink 1.4s infinite both;
          opacity: 0;
        }
        
        .dot:nth-child(1) { animation-delay: 0s; }
        .dot:nth-child(2) { animation-delay: 0.4s; }
        .dot:nth-child(3) { animation-delay: 0.8s; }
        
        @keyframes blink {
          0%, 40%   { opacity: 0; }
          60%       { opacity: 1; }
          100%      { opacity: 0; }
        }
      </style>
    </head>
    <body>
      <h1>Looking for <span class="highlight">${keyword}</span> in our server
        <span class="loading-dots">
          <span class="dot">.</span>
          <span class="dot">.</span>
          <span class="dot">.</span>
        </span>
      </h1>
    </body>
    </html>
  `);
};

export const userLogin = (req, res) => {
  res.send(
    '<h1>This is the <span style="color: green;">Login</span> Route!</h1>'
  );
};

export const userSignUp = (req, res) => {
  res.send(
    '<h1>This is a <span style="color: green;">Signup</span> Route!</h1>'
  );
};

export const usersHandler = (req, res) => {
  const { name, email } = req.body;
  res.json({
    message: `Account under name ${name} and email ${email} created successfully!`,
  });
};

export const changeHandler = (req, res) => {
  const userID = req.params.id;
  const { name, email } = req.body;
  res.json({
    message: `User ${userID} updated to name ${name} and email ${email}`,
  });
};

export const deleteHandler = (req, res) => {
  const userID = req.params.id;
  res.json({
    message: `Account under user ${userID} has been deleted successfully!`,
  });
};

export const staffHandler = (req, res) => {
  const { name, id } = req.params;

  if (!/^\d{12}$/.test(id)) {
    return res.status(400).json({ error: "id must be exactly 12 digits!" });
  }

  res.json({ id, name });
};

export const bookHandler = (req, res) => {
  const book = req.params.book;
  res.send(
    `<!DOCTYPE html>
    <html lang="en">
    <head>
      <meta charset="UTF-8">
      <title>Searching...</title>
      <style>
        .highlight{
            color: green;
        }
        .loading-dots {
          display: inline-block;
          min-width: 3ch;
        }
        
        .dot {
          animation: blink 1.4s infinite both;
          opacity: 0;
        }
        
        .dot:nth-child(1) { animation-delay: 0s; }
        .dot:nth-child(2) { animation-delay: 0.4s; }
        .dot:nth-child(3) { animation-delay: 0.8s; }
        
        @keyframes blink {
          0%, 40%   { opacity: 0; }
          60%       { opacity: 1; }
          100%      { opacity: 0; }
        }
      </style>
    </head>
    <body>
      <h1>Searching for the book "<span class="highlight">${book}</span>" in our server
        <span class="loading-dots">
          <span class="dot">.</span>
          <span class="dot">.</span>
          <span class="dot">.</span>
        </span>
      </h1>
    </body>
    </html>`
  );
};
